sdk/cloudmachine/Azure.Projects.OpenAI/src/AIServiceFeature.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Projects.Core;
+using Azure.Provisioning.CognitiveServices;
+
+namespace Azure.Projects.OpenAI;
+
+/// <summary>
+/// AIServices feature.
+/// </summary>
+public class AIServicesFeature : AzureProjectFeature
+{
+    private string _model;
+    private string _modelVersion;
+
+    /// <summary>
+    /// Creates a new instance of the AIServices feature.
+    /// </summary>
+    /// <param name="model"></param>
+    /// <param name="modelVersion"></param>
+    public AIServicesFeature(string model, string modelVersion)
+    {
+        _model = model;
+        _modelVersion = modelVersion;
+    }
+
+    /// <summary>
+    /// Adds the feature to the project infrastructure.
+    /// </summary>
+    /// <param name="infrastructure"></param>
+    protected override void EmitConstructs(ProjectInfrastructure infrastructure)
+    {
+        CognitiveServicesAccount cognitiveServices = new("aiservices")
+        {
+            Name = infrastructure.ProjectId,
+            Kind = "AIServices",
+            Sku = new CognitiveServicesSku { Name = "S0" },
+            Properties = new CognitiveServicesAccountProperties()
+            {
+                PublicNetworkAccess = ServiceAccountPublicNetworkAccess.Enabled,
+                CustomSubDomainName = infrastructure.ProjectId,
+                //DisableLocalAuth = true,
```

Comment:
intentional?

Reply:
yes, but I was considering turning it off. should we?

Reply:
We should if we can get away with it - this just makes the deployment more secure by only allowing Entra ID.

Reply:
will fix